### PR TITLE
Fix astropy so that it can be used in app bundles, and add CI to prevent regression

### LINKED ISSUE #7052: Error building executable using pyinstaller
Building executables which include `astropy` seem to have a history of issues (e.g. https://github.com/astropy/astropy/pull/960#issuecomment-355678708).  Initially I was getting an,
`ImportError: Astropy requires the 'six' module of minimum version 1.10`, which I solved by using a trick given in the above PR: 

> I was able to get this to work using PyInstaller with the following pyinstaller options` --exclude-module astropy` and `--add-data path_to/astropy:astropy` -- @fridgerator 

Now I get the new error:

```
Traceback (most recent call last):
  File "pyapi.py", line 10, in <module>
    import cosmopy
  File "/Users/lzkelley/Programs/cosmo/ve_test/vetest/lib/python3.5/site-packages/PyInstaller/loader/pyimod03_importers.py", line 631, in exec_module
    exec(bytecode, module.__dict__)
  File "site-packages/cosmopy/__init__.py", line 9, in <module>
  File "/var/folders/wr/_dhmn18x1dj6t8ytjjd9rsxh0000gn/T/_MEIhGawWS/astropy/constants/__init__.py", line 28, in <module>
    from .constant import Constant, EMConstant
  File "/var/folders/wr/_dhmn18x1dj6t8ytjjd9rsxh0000gn/T/_MEIhGawWS/astropy/constants/constant.py", line 11, in <module>
    from ..units.core import Unit, UnitsError
  File "/var/folders/wr/_dhmn18x1dj6t8ytjjd9rsxh0000gn/T/_MEIhGawWS/astropy/units/__init__.py", line 14, in <module>
    from .core import *
  File "/var/folders/wr/_dhmn18x1dj6t8ytjjd9rsxh0000gn/T/_MEIhGawWS/astropy/units/core.py", line 23, in <module>
    from .utils import (is_effectively_unity, sanitize_scale, validate_power,
  File "/var/folders/wr/_dhmn18x1dj6t8ytjjd9rsxh0000gn/T/_MEIhGawWS/astropy/units/utils.py", line 16, in <module>
    from fractions import Fraction
ImportError: No module named 'fractions'
[2338] Failed to execute script pyapi
```

I'm using a virtualenv named `vetest` in the directory `ve_test`, trying to run with the command:
`pyinstaller -F pyapi.py --exclude-module astropy --add-data /Users/lzkelley/Programs/cosmo/ve_test/vetest/lib/python3.5/site-packages/astropy:astropy`

The full output from `pyinstaller` is attached.

[out.txt](https://github.com/astropy/astropy/files/1608048/out.txt)

---------

An additional issue is that the resulting executable is >200MB, is that normal?  The application I'm trying to package is really just using the `astropy.cosmology.LambdaCDM` class.  Might there be a better way to include that alone?

### PR DESCRIPTION
**Pull request by astrofrog**
With applications like [pyinstaller](https://www.pyinstaller.org/), it's easy to build application bundles for different platforms. Unfortunately, astropy doesn't work out of the box with these and requires some tweaking due to various issues. In the past, @embray actually opened a PR to fix this (https://github.com/astropy/astropy/pull/960) but this was never finished.

Since it would be nice for this to work cleanly, I've decided to start off by adding a CircleCI job that will test that this works correctly and that we don't regress in future. I'll then work on fixing issues, and will see if I can cherry pick or adapt any of the solutions from https://github.com/astropy/astropy/pull/960. So this is just a WIP for now.

UPDATE: this is now ready, see https://github.com/astropy/astropy/pull/8795#issuecomment-623552977

EDIT: Fix #7052

### CODE REVIEW

**Comment by astrofrog** [comment]
Some of the fixes will much easier to do by adding a ``hook-astropy.py`` into PyInstaller, so I'll work on that.

**Comment by bsipocz** [comment]
@astrofrog - I'm bumping the milestone to the next bugfix, to help us see how we stand. Of course if any from there is ready before the release, they can be moved back to 4.0.

**Comment by astrofrog** [comment]
Sounds good, I definitely won't have time to finish this for 4.0.

**Comment by bsipocz** [comment]
is this still something on the table post APE17?

**Comment by astrofrog** [comment]
Yes this would still be nice to get working, I just haven't had time

**Comment by bsipocz** [comment]
no worries, I just remilestoned it then. We can always move it back to the bugfix milestone when it's merged.

**Comment by mhvk** [comment]
Not release-critical, so removing milestone.

**Comment by astrofrog** [comment]
This is now ready for review - all required fixes to make astropy work properly in bundles were done in separate PRs over the last year or so, this now just add a tox environment and associated Travis cron job to make sure that we don't break this capability. The Travis job is 7 minutes in total, so not a huge overhead, and I've only enabled it as a cron job to avoid slowing down any PRs. I had a temporary commit here to make sure it ran correctly, but have now removed the commit so the job won't show up here (and CI should therefore pass anyway).

A side bonus is that this makes sure that the tests are able to be extracted as-is and run against an installed version of astropy (this was needed since pytest cannot discover tests in application bundles) - normally this should work, but there were a few relative imports that have crept in since we switched to absolute imports. Relative imports to the same directory are fine, but this fixes ones where the relative import went one or more levels up (which shouldn't be there).

In any case, this is now ready for review and can be backported to 4.0.x.

**Comment by astrofrog** [comment]
I think with this merged we can close https://github.com/astropy/astropy/issues/7052

File: .travis.yml [hunk 1]
```diff
@@ -188,6 +188,14 @@ matrix:
           stage: Cron tests
           env: SETUP_METHOD='apt'
 
+        # Regularly make sure that astropy can be used
```
Comment (mhvk) [inline .travis.yml:191]: Comment incomplete - "in an Application bundle"?

Comment (astrofrog) [reply .travis.yml:191]: Oops thanks :)

**Comment by astrofrog** [comment]
The success over in #7052 is due to fixes that were done in other PRs in the process of doing this one - this current PR essentially just adds the regression test :smile:

**Comment by astrofrog** [comment]
Since this has been approved, I'll go ahead and merge so that it doesn't go stale :smile:

**Comment by embray** [comment]
Wow, my first attempt to solve this was 7 years ago now.  Amazing to see this finally done :)

**Comment by astrofrog** [comment]
@embray thank you for starting this!

**Pull request merged**